client/src/telemetry/glean-context.tsx

```diff
@@ -0,0 +1,92 @@
+import * as React from "react";
+import { path, referrer } from "./generated/pageView";
+import { clicked } from "./generated/element";
+import * as pings from "./generated/pings";
+import * as Glean from "@mozilla/glean/web";
+import { GLEAN_CHANNEL, GLEAN_DEBUG, GLEAN_ENABLED } from "../env";
+
+export type PageProps = {
+  referrer: string | undefined;
+  path: string | undefined;
+};
+
+export type PageEventProps = {
+  referrer: string | undefined;
+  path: string | undefined;
+};
+
+export type ElementClickedProps = {
+  source: string;
+  subscription_type: string;
+};
+
+export type GleanAnalytics = {
+  page: (arg: PageProps) => void;
+  click: (arg: ElementClickedProps) => void;
+};
+
+const FIRST_PARTY_DATA_OPT_OUT_COOKIE_NAME = "moz-1st-party-data-opt-out";
+const GLEAN_APP_ID = "mdn-yari";
+
+function glean(): GleanAnalytics {
+  if (typeof window === "undefined") {
```

Comment:
I learned in https://github.com/mdn/yari/pull/6295 that we shouldn't use this to detect SSR.

Can we use the `useIsServer()` hook here?

Reply:
We cannot it is outside of a React function component.